src/components/CommentForm.jsx

```diff
@@ -0,0 +1,76 @@
+import Button from "@mui/material/Button";
+import Snackbar from "@mui/material/Snackbar";
+import MuiAlert from "@mui/material/Alert";
+import * as React from "react";
+import { useContext, useState } from "react";
+import { postArticleComment } from "../utils";
+import { UserContext } from "../contexts/UserContext";
+import { TextField } from "@mui/material";
+
+const Alert = React.forwardRef(function Alert(props, ref) {
+  return <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />;
+});
+
+export default function CommentForm({ article_id, addComment }) {
+  const { user } = useContext(UserContext);
+  const [body, setBody] = useState("");
+  const [messageOpen, setMessageOpen] = useState(false);
+  const [message, setMessage] = useState("");
+  const [error, setError] = useState(false);
+  const handleSubmit = () => {
+    if (body.length === 0) {
+      setError(true);
+      return;
+    }
+    const comment = { username: user.username, body };
+    postArticleComment(article_id, comment)
+      .then(({ comment }) => {
+        addComment(comment);
```

Comment:
Good to see some optimistic rendering here if the comment post is successful, and without making a request to get the whole single article from the api again.